libcodechecker/analyze/log_parser.py

```diff
@@ -55,78 +55,40 @@ def get_compiler_includes(compiler, extra_opts=None):
             if line.startswith(end_mark):
                 do_append = False
             if do_append:
-                include_paths.append("-I"+line)
+                include_paths.append("-I" + line)
             if line.startswith(start_mark):
                 do_append = True
 
     except OSError as oerr:
-        LOG.error("Cannot find include paths:" + oerr.strerror+"\n")
+        LOG.error("Cannot find include paths: " + oerr.strerror + "\n")
     return include_paths
 
 
-# Certain includes are defined by GCC to a GCC internal "macro" never exported
-# by the compiler, such as
-# #define __has_include(STR) __has_include__(STR)
-# which causes Clang to fail in certain cross-compiler and cross-architecture
-# setups due to blabla__ not being defined by Clang in a way GCC does it.
-#
-# Thus, we ignore these "defines" and let Clang fall back to its own
-# __has_include definition. List taken from the "extension" list of Clang and
-# the GCC extension list referenced therein:
-# https://clang.llvm.org/docs/LanguageExtensions.html
-IGNORED_DEFINES = frozenset(
-    ['__has_include'])  # Matches __has_include and __has_include_next.
-
-
-def get_compiler_defines(compiler, extra_opts=None):
+def get_compiler_target(compiler):
```

Comment:
if the compiler is not gcc return an empty string.
For example for clang compilers this function will always return the host architecture as opposed to the build target.